### PR TITLE
fix(common): evaluate ingress secret name before condition

### PR DESCRIPTION
**Pull request by ammmze**
### Description of the change
Evaluates the ingress TLS secret name as a template before deciding whether to include or or omit the `secretName` field.

#### Fixed
Using an ingress TLS secret name template evaluating to an empty string resulted in the `secretName` field still being added to the ingress TLS config with an empty string value.

### Benefits
Allows using a template that may or may not cause the `secretName` field to be present in the Ingress TLS config.

### Possible drawbacks
I guess this takes away the ability to ever add the ingress TLS secret name as an empty string. But I'm not sure ever having that as empty string is a valid use case.

### Applicable issues
None

## Checklist

- [x] Title of the PR conforms to the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard.
- [x] Scope of the of the PR title contains the chart name.
- [x] Chart version in `Chart.yaml` has been bumped according to [Semantic Versioning](https://semver.org/).
- [x] Chart `artifacthub.io/changes` changelog annotation has been updated in `Chart.yaml`. See [Artifact Hub documentation](https://artifacthub.io/docs/topics/annotations/helm/#supported-annotations) for more info.
- [ ] ~Variables have been documented in the `values.yaml` file.~

### CODE REVIEW

File: charts/library/common/templates/classes/_ingress.tpl [hunk 1]
```diff
@@ -41,12 +41,13 @@ spec:
   {{- if $values.tls }}
   tls:
     {{- range $values.tls }}
+    {{- $secretName := tpl .secretName $ }}
```
Comment (ammmze) [inline charts/library/common/templates/classes/_ingress.tpl:44]: I guess we need to verify this is a string before pass it to the template engine.

Comment (ammmze) [reply charts/library/common/templates/classes/_ingress.tpl:44]: Perhaps something like this?

```suggestion
    {{- $secretName := .secretName | ternary (tpl .secretName $) "" }}
```

Comment (bjw-s) [reply charts/library/common/templates/classes/_ingress.tpl:44]: `ternary` makes stuff so hard to read, I like `default` a bit better 😂
```suggestion
    {{- $secretName := tpl (default "" .secretName) $ }}
```

**Review by bjw-s — CHANGES_REQUESTED** [review]

File: charts/library/common/templates/classes/_ingress.tpl (hunk 1 above)
Comment (ammmze) [reply charts/library/common/templates/classes/_ingress.tpl]: 👍🏻 You got it! Changes have been pushed.

**Review by bjw-s — APPROVED** [review]

**Pull request merged**